engine/runtime/src/test/scala/org/enso/compiler/test/pass/desugar/GenerateMethodBodiesTest.scala

```diff
@@ -248,7 +250,15 @@ class GenerateMethodBodiesTest extends CompilerTest {
     }
 
     "return an error when redefining `self` parameter" in {
-      irResultQux.body shouldBe an[errors.Redefined.SelfArg]
+      irResultQux.body shouldBe a[Function]
```

Comment:
I'd prefer a test in Java, not Scala. For example in `ErrorCompilerTest` which contains `assertSingleSyntaxError`.